examples/impute/plot_iterative_imputer_as_missforest.py

```diff
@@ -0,0 +1,61 @@
+"""
+=============================================================
+Replicating Functionality of missForest with IterativeImputer
+=============================================================
+
+There are many well-established imputation packages in the R data science
+ecosystem: Amelia, mi, mice, missForest, and others.
+
+missForest is popular, and turns out to be a particular instance of a class of
+sequential imputation algorithms that can all be implemented with the
+:class:`sklearn.impute.IterativeImputer` class, which is a strategy for
+imputing missing values by modeling each feature with missing values as a
+function of other features in a round-robin fashion. In the case of missForest,
+the function is a Random Forest.
+
+In this example we will demonstrate how to use
+:class:`sklearn.impute.IterativeImputer` to replicate the functionality of
+missForest.
+"""
+print(__doc__)
+
+import numpy as np
+
+from sklearn.datasets import load_diabetes
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.impute import IterativeImputer
+
+rng = np.random.RandomState(0)
+
+# Load data
+dataset = load_diabetes()
+X_full, y_full = dataset.data, dataset.target
+n_samples = X_full.shape[0]
+n_features = X_full.shape[1]
+
+# Add missing values in 75% of the lines
+missing_rate = 0.75
+n_missing_samples = int(np.floor(n_samples * missing_rate))
+missing_samples = np.hstack((np.zeros(n_samples - n_missing_samples,
+                                      dtype=np.bool),
+                             np.ones(n_missing_samples,
+                                     dtype=np.bool)))
+rng.shuffle(missing_samples)
+missing_features = rng.randint(0, n_features, n_missing_samples)
+
+X_missing = X_full.copy()
+X_missing[np.where(missing_samples)[0], missing_features] = np.nan
+y_missing = y_full.copy()
+
+# Random Forest predictor with default values according to missForest docs
+predictor = RandomForestRegressor(n_estimators=100, max_features='sqrt')
```

Comment:
So this uses a different random state in each imputation round. What if we want it to be deterministic? Is it a problem then if we use the exact same random state (e.g. random_state=0) in each imputation round? Is there a way to make each RF have a different, fixed random state???

Reply:
(What does missForest say about random determinism?)

Reply:
I can change this to be `random_state=0` for reproducibility. I can't think of any problem of using the same random state in each imputation round since missForest is not like MICE. I.e., we are not sampling from posteriors.

There is nothing in the `missForest` paper about random determinism - I searched the PDF for variants of both of those words and nothing relevant came up https://arxiv.org/abs/1105.0828